src/Select/index.js

```diff
@@ -0,0 +1,149 @@
+  onChange = noop,
+  defaultValue,
+  defaultOpen = false,
+  errorText,
+}) => {
+  // only include valid Select children in options items
+  const items = React.Children.toArray(children).filter((child) =>
+    ["SelectItem", "SelectAction"].includes(child.type.name)
+  );
+
+  const defaultSelectedItem = items
+    .filter((item) => !isAction(item)) // action items may not be selected by default
+    .filter((item) => item.props.value === defaultValue)
+    .pop();
+
+  /** @see https://www.downshift-js.com/use-select */
+  const {
+    isOpen,
+    selectedItem,
+    getToggleButtonProps,
+    getLabelProps,
+    getMenuProps,
+    highlightedIndex,
+    getItemProps,
+  } = useSelect({
+    items,
+    defaultSelectedItem,
+    initialIsOpen: defaultOpen,
+    itemToString: (item) => item.props.value || item.props.children, // typeahead string
+    onSelectedItemChange: ({ selectedItem }) => {
+      // for Select.Action items, we only fire the side effect
+      if (isAction(selectedItem)) {
+        selectedItem.props.onSelect();
+      } else {
+        onChange(selectedItem.props.value);
+      }
+    },
+  });
+
+  return (
+    <div className="nds-select">
+      <DropdownTrigger
+        isOpen={isOpen}
+        labelText={label}
+        displayValue={getSelectedItemDsiplay(selectedItem)}
+        labelProps={{ ...getLabelProps() }}
+        errorText={errorText}
+        {...getToggleButtonProps()}
+      />
+      <ul
+        className={cc([
+          "nds-select-list",
+          "list--reset bgColor--white rounded--all border--all",
+          { "nds-select-list--active": isOpen },
+        ])}
+        {...getMenuProps()}
+      >
+        {isOpen &&
+          items.map((item, index) => (
+            <li
+              key={`${item}${index}`}
+              className={cc([
+                "nds-select-item",
+                "alignChild--left--center padding--x--s padding--y--xs",
+                {
+                  "nds-select-item--highlighted": highlightedIndex === index,
+                  "rounded--top": index === 0,
+                  "rounded--bottom": index === items.length - 1,
+                },
+              ])}
+              {...getItemProps({ item, index })}
+            >
+              {item}
+            </li>
+          ))}
+      </ul>
+    </div>
+  );
+};
+
+Select.propTypes = {
+  /** Label for the select control */
+  label: PropTypes.string.isRequired,
+  /** Change callback. Called with value string from the selected item */
+  onChange: PropTypes.func,
+  /**
+   * Use to set a default selection by passing the `value` prop
+   * of one of the `<Select.Item>` children.
+   */
+  defaultValue: PropTypes.string,
+  /** Open the dropdown on rendder if `true` */
+  defaultOpen: PropTypes.bool,
+  /**
+   * Error message.
+   * When passed, this will cause the trigger to render in error state.
+   */
+  errorText: PropTypes.string,
+};
+
+Select.Item = SelectItem;
+Select.Action = SelectAction;
```

Comment:
neat! 📸

Reply:
we do the same thing in `Tabs` so that `import { Tabs } from '@narmi/design_system';` pulls in the main component and its subcomponents like `Tabs.Panel`.